### PR TITLE
NIMBUS-21215:: new, update, delete actions for remote entities

### PR DESCRIPTION
**Pull request by rakeshnpatel**
# Description
The changes include support for _new, _update, _delete actions (along with _get and _search) for remote configured entities via REST calls
The response for remote invocation would always be raw state of the entity (similar to $state behavior)

# Overview of Changes
1. RemoteWSModelRepository now implements _new, _update and _delete
2. Added new ResponseInterceptor for when the request contains "responseBody=_raw". this is used for all the REST calls on remote entities.
3. More details will be available in the documentation

# Type of Change
- [ ] New feature
- [ ] Refactor/Technical Debt
- [ ] Test case change
- [ ] This change requires a documentation update

# Test Details
ModelRepositoryTest class contains the unit tests for these actions on remote configured entity

# Additional Notes
TODO changes:
1. For process state persistence, added a separate ProcessRepository for MongoDB that contains operations in the earlier format to continue supporting CRUDs. Need to re-design the approach and accordingly implementation. Reviewing with @jayantchaudhuri 
2. Need to handle executeError in response (where result is null) in the response for remote entities rest calls - in progress and will be a separate PR
3. Need to address how service urls for rest calls on remote entities are being looked up from yml, configuring urls at each domain alias level seems redundant. - TBD

### CODE REVIEW

**Comment by openanthem-admin** [comment]
You did it @rakeshnpatel!

Thank you for signing the Anthem Opensource Software CLA.

File: nimbus-core/src/main/java/com/antheminc/oss/nimbus/channel/web/RemoteModelClientHttpRequestInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+			super(request);
+		}
+		
+		@Override
+		public URI getURI() {
+			URI uri = super.getURI();
+			
+			String commandUri = uri.toString();
+			
+			//Override behavior to $execute for remote model ws calls
```
[the remark this answers is not in the document]
Comment (rakeshnpatel) [reply nimbus-core/src/main/java/com/antheminc/oss/nimbus/channel/web/RemoteModelClientHttpRequestInterceptor.java]: I think CommandBuilder is mostly used to take the "url" string and convert it into a command object where query params are converted into command params and so on... and that command is what we use going forward.

In this case, it is intercepting an HTTP request and trying to change the requestUri (may be variable name commandUri is misleading ?) to change behavior query param to just include "$execute". e,g, 
requestUri = http://localhost:8080/.../p/abc/_search?b=$executeAnd$state&param1=test
to
requestUri = http://localhost:8080/.../p/abc/_search?b=$execute&param1=test 

so thinking it is just a requestUri string update and don't really need to create command etc.. and hence put it here as string replace. 

Please let me know.

[the remark this answers is not in the document]
Comment (soham-anthem) [reply nimbus-core/src/main/java/com/antheminc/oss/nimbus/channel/web/RemoteModelClientHttpRequestInterceptor.java]: I would consider viewing all these scenarios as variations of Command. For example, here we could think that its Command being passed in, just that its translated in form of String in this layer. So, whenever we start working with it, we first objectify it to Command then work. From f/w perspective, it is not dealing with HTTP request or JMS/Queue/Topic Messages or SMTP request or any other protocols - we can uniformly represent it as Command when working with it. Command will have different representations based on protocol in which it is used.

[the remark this answers is not in the document]
Comment (rakeshnpatel) [reply nimbus-core/src/main/java/com/antheminc/oss/nimbus/channel/web/RemoteModelClientHttpRequestInterceptor.java]: Moved the uri update to command class. please review

**Pull request merged**